src/cursor/cur_table.c

```diff
@@ -511,9 +511,16 @@ __curtable_insert(WT_CURSOR *cursor)
 	 */
 	F_SET(primary, flag_orig | WT_CURSTD_KEY_EXT | WT_CURSTD_VALUE_EXT);
 
-	if (ret == WT_DUPLICATE_KEY && F_ISSET(cursor, WT_CURSTD_OVERWRITE))
+	if (ret == WT_DUPLICATE_KEY && F_ISSET(cursor, WT_CURSTD_OVERWRITE)) {
 		WT_ERR(__curtable_update(cursor));
-	else {
+
+		/*
+		 * The cursor is no longer positioned. This isn't just cosmetic,
+		 * without a reset, iteration on this cursor won't start at the
+		 * beginning/end of the table.
+		 */
+		APPLY_CG(ctable, reset);
```

Comment:
Yes, that's a fair point.

This change means that after an `insert` (where the key already exists), a subsequent `search/search_near` will always do a full tree search rather than first checking a currently pinned leaf page.

We could document that `update` is a better choice than `insert` when doing repeated, localized, paired search and update operations, but I'm not sure exactly what that wording would look like.